src/url-state-machine.js

```diff
@@ -93,6 +93,20 @@ function percentEncode(c) {
   return "%" + hex;
 }
 
+function percentDecode(input) {
```

Comment:
This *is* actually a byte sequence -> byte sequence operation, and the change is part of my attempt to get the module more aligned with the spec.